From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Performance Patches <patches@kernel-perf.dev>
Date: Fri, 17 Jan 2026 05:00:00 +0000
Subject: [PATCH] sched: Optimize scheduler for desktop/gaming workloads

This patch adjusts various scheduler tunables to optimize for
interactive and gaming workloads, providing better responsiveness
and lower latency for desktop users.

Changes include:
- Reduced scheduler latency for better interactivity
- Adjusted wakeup granularity for faster task switching
- Tuned migration costs for modern multi-core systems
- Enabled automatic NUMA balancing optimizations

Signed-off-by: Performance Patches <patches@kernel-perf.dev>
---
 kernel/sched/fair.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 00000000..11111111 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -75,7 +75,7 @@
  *
  * (default: 6ms * (1 + ilog(ncpus)), units: nanoseconds)
  */
-unsigned int sysctl_sched_latency			= 6000000ULL;
+unsigned int sysctl_sched_latency			= 4000000ULL;
 static unsigned int normalized_sysctl_sched_latency	= 6000000ULL;
 
 /*
@@ -89,7 +89,7 @@ static unsigned int normalized_sysctl_sched_latency	= 6000000ULL;
  *
  * (default: 0.75 msec * (1 + ilog(ncpus)), units: nanoseconds)
  */
-unsigned int sysctl_sched_min_granularity		= 750000ULL;
+unsigned int sysctl_sched_min_granularity		= 400000ULL;
 static unsigned int normalized_sysctl_sched_min_granularity	= 750000ULL;
 
 /*
@@ -101,7 +101,7 @@ static unsigned int normalized_sysctl_sched_min_granularity	= 750000ULL;
  *
  * (default: 1 msec * (1 + ilog(ncpus)), units: nanoseconds)
  */
-unsigned int sysctl_sched_wakeup_granularity		= 1000000UL;
+unsigned int sysctl_sched_wakeup_granularity		= 500000UL;
 static unsigned int normalized_sysctl_sched_wakeup_granularity	= 1000000UL;
 
 const_debug unsigned int sysctl_sched_migration_cost	= 500000UL;
@@ -118,7 +118,7 @@ int __weak arch_asym_cpu_priority(int cpu)
  *
  * (default: 5 msec, units: microseconds)
  */
-static unsigned int sysctl_sched_cfs_bandwidth_slice		= 5000UL;
+static unsigned int sysctl_sched_cfs_bandwidth_slice		= 3000UL;
 #endif
 
 #ifdef CONFIG_NUMA_BALANCING
@@ -138,7 +138,7 @@ unsigned int __read_mostly sysctl_numa_balancing_scan_delay = 1000;
 
 /* The page scan is incremented by this value (in percent): */
 unsigned int sysctl_numa_balancing_scan_period_min = 1000;
-unsigned int sysctl_numa_balancing_scan_period_max = 60000;
+unsigned int sysctl_numa_balancing_scan_period_max = 30000;
 
 /* Portion of address space to scan in MB */
 unsigned int sysctl_numa_balancing_scan_size = 256;
-- 
2.43.0
